Core/src/org/sleuthkit/autopsy/ingest/IngestJob.java

```diff
@@ -204,8 +203,13 @@ private IngestJobPipeline getStreamingIngestPipeline() {
 	 * Set up the pipeline(s)
 	 */
 	if (ingestMode == Mode.STREAMING) {
-	    IngestJobPipeline ingestJobPipeline = new IngestJobPipeline(this, streamingIngestDataSource, settings);
-	    this.ingestJobPipelines.put(ingestJobPipeline.getId(), ingestJobPipeline);
+            if (streamingIngestDataSource == null) {
+                logger.log(Level.SEVERE, "Error starting streaming ingest - data source not set");
+                return new ArrayList<>();
```

Comment:
I think that we need to return a non-empty list or otherwise signal that there was an error and the job won't be executing. Then we can leaving logging to the higher level client code, as has been the case before. Maybe we need a start(DataSource) overload after all?

Reply:
This code will change once the DSP gets moved (it won't be possible for the data source to be missing), so I don't think it's worth doing any major changes right now.

Reply:
OK